test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommandTests.cs

```diff
@@ -4137,6 +4138,35 @@ await SimpleTestPackageUtility.CreateFolderFeedV3Async(
             result.LockFile.Targets[0].Libraries[1].Build.Should().BeEquivalentTo([new LockFileItem("build/net45/_._")]);
         }
 
+        [Fact]
+        public async Task Restore_WithHttpsSourceIndexJsonReturningHttpResources_LogsNU1303()
+        {
+            // Arrange
+            using var pathContext = new SimpleTestPathContext();
+
+            // Start HTTPS mock server with index.json returning HTTP resources
+            using var server = new SelfSignedHttpsServerWithHttpResources();
+            var serverTask = server.StartServerAsync();
+
+            var httpsSource = server.URI + "v3/index.json";
+            var logger = new TestLogger();
+            pathContext.Settings.AddSource("https-feed", httpsSource, "disableTLSCertificateValidation", "true");
+            var settings = Settings.LoadDefaultSettings(pathContext.SolutionRoot);
+            var project1Spec = ProjectTestHelpers.GetPackageSpec(settings, "Project1", pathContext.SolutionRoot, framework: "net5.0");
+            AddDependency(project1Spec, "SomePackage", "1.0.0");
+
+            var request = ProjectTestHelpers.CreateRestoreRequest(pathContext, logger, project1Spec);
+            var command = new RestoreCommand(request);
+
+            // Act
+            var result = await command.ExecuteAsync();
+
+            // Assert
+            server.StopServer();
+            result.Success.Should().BeFalse(because: logger.ShowMessages());
+            result.LockFile.LogMessages.Should().ContainSingle(m => m.Code == NuGetLogCode.NU1302 && m.Message.Contains("http://"));
```

Comment:
nit: In a follow-up, can the resource string be referenced here instead of "http://" ?